Tribler/community/market/core/orderbook.py

```diff
@@ -0,0 +1,404 @@
+        return profile
+
+    def get_ask_side_depth_profile(self, price_wallet_id, quantity_wallet_id):
+        """
+        format: [(<price>, <depth>), (<price>, <depth>), ...]
+
+        :return: The depth profile
+        :rtype: list
+        """
+        profile = []
+        for key, value in self._asks.get_price_level_list(price_wallet_id, quantity_wallet_id).items():
+            profile.append((key, value.depth))
+        return profile
+
+    def bid_relative_price(self, price):
+        """
+        :param price: The price to be relative to
+        :type price: Price
+        :return: The relative price
+        :rtype: Price
+        """
+        assert isinstance(price, Price), type(price)
+        return self.get_bid_price('BTC', 'MC') - price
+
+    def ask_relative_price(self, price):
+        """
+        :param price: The price to be relative to
+        :type price: Price
+        :return: The relative price
+        :rtype: Price
+        """
+        assert isinstance(price, Price), type(price)
+        return self.get_ask_price('BTC', 'MC') - price
+
+    def relative_tick_price(self, tick):
+        """
+        :param tick: The tick with the price to be relative to
+        :type tick: Tick
+        :return: The relative price
+        :rtype: Price
+        """
+        assert isinstance(tick, Tick), type(tick)
+
+        if tick.is_ask():
+            return self.ask_relative_price(tick.price)
+        else:
+            return self.bid_relative_price(tick.price)
+
+    def get_bid_price_level(self, price_wallet_id, quantity_wallet_id):
+        """
+        Return the price level that an ask has to match to make a trade
+        :rtype: PriceLevel
+        """
+        return self._bids.get_max_price_list(price_wallet_id, quantity_wallet_id)
+
+    def get_ask_price_level(self, price_wallet_id, quantity_wallet_id):
+        """
+        Return the price level that a bid has to match to make a trade
+        :rtype: PriceLevel
+        """
+        return self._asks.get_min_price_list(price_wallet_id, quantity_wallet_id)
+
+    def get_order_ids(self):
+        """
+        Return all IDs of the orders in the orderbook, both asks and bids. The returned list is sorted.
+
+        :rtype: [OrderId]
+        """
+        ids = []
+
+        for price_wallet_id, quantity_wallet_id in self.asks.get_price_level_list_wallets():
+            for _, price_level in self.asks.get_price_level_list(price_wallet_id, quantity_wallet_id).items():
+                for ask in price_level:
+                    ids.append(ask.tick.order_id)
+
+        for price_wallet_id, quantity_wallet_id in self.bids.get_price_level_list_wallets():
+            for _, price_level in self.bids.get_price_level_list(price_wallet_id, quantity_wallet_id).items():
+                for bid in price_level:
+                    ids.append(bid.tick.order_id)
+
+        return sorted(ids)
+
+    def __str__(self):
+        from cStringIO import StringIO
```

Comment:
1.) Why cStringIO: this is not available on all platforms, please use the StringIO package, 2.) Why use StringIO here at all?